### PR TITLE
[[ Bug 15571 ]] Fix UTF-32 encoding - memory corruption and failure i…

### PR DESCRIPTION
**Pull request by livecodesebastien**
…n encoding surrogate pairs

### CODE REVIEW

File: libfoundation/src/foundation-string.cpp [hunk 1]
```diff
@@ -350,7 +350,10 @@ bool MCStringCreateWithBytes(const byte_t *p_bytes, uindex_t p_byte_count, MCStr
 				}
 				else
 				{
-					// Split to surrogate pairs
+                    // Split to surrogate pairs
+                    // SN-2015-07-03: [[ Bug 15571 ]] Creating a surrogate pair
+                    //  makes the UTF-16 string longer.
+                    t_buffer . Extend(t_buffer . Size() + 1);
```
Comment (runrevmark) [inline libfoundation/src/foundation-string.cpp]: This can fail due to OOM - check the return value and return false if necessary.

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 4d56d8a

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 4d56d8a43301c571b5bb7581518bfe5a51f9571b

**Comment by livecode-vulcan** [comment]
😞 test failure 4d56d8a43301c571b5bb7581518bfe5a51f9571b
- try-linux-x86: [success](https://vulcan.livecode.com/builders/try-linux-x86/builds/421)
- try-win-x86: [failure](https://vulcan.livecode.com/builders/try-win-x86/builds/481)
- try-ios: [success](https://vulcan.livecode.com/builders/try-ios/builds/367)
- try-android-armv6: [success](https://vulcan.livecode.com/builders/try-android-armv6/builds/432)
- try-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/367)
- try-mac: [success](https://vulcan.livecode.com/builders/try-mac/builds/358)

**Comment by livecodesebastien** [comment]
@livecode-vulcan retry 4d56d8a

**Comment by livecode-vulcan** [comment]
😎 test success 4d56d8a43301c571b5bb7581518bfe5a51f9571b
- try-linux-x86: [success](https://vulcan.livecode.com/builders/try-linux-x86/builds/422)
- try-win-x86: [success](https://vulcan.livecode.com/builders/try-win-x86/builds/482)
- try-ios: [success](https://vulcan.livecode.com/builders/try-ios/builds/368)
- try-android-armv6: [success](https://vulcan.livecode.com/builders/try-android-armv6/builds/433)
- try-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/368)
- try-mac: [success](https://vulcan.livecode.com/builders/try-mac/builds/359)

**Pull request merged**